File: src/pt/joaomneto/titancompanion/adventure/impl/fragments/aod/SoldiersDivision.java
package pt.joaomneto.titancompanion.adventure.impl.fragments.aod;

/**
 * Created by joao on 26/04/17.
 */

public class SoldiersDivision {
    private String type;
    private int quantity;

    public SoldiersDivision(String type, int quantity) {
        this.type = type;
        this.quantity = quantity;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/fragments/aod/SoldiersDivision.kt
package pt.joaomneto.titancompanion.adventure.impl.fragments.aod

import android.content.Context

/**
 * Created by 962633 on 13-09-2017.
 */
interface SoldiersDivision {

    val quantity: Int
    val initialQuantity: Int
    val category: String

    fun resetToInitialValues(): SoldiersDivision
    fun incrementAllValues(): SoldiersDivision
    fun decrementAllValues(): SoldiersDivision
    fun getLabel(ctx: Context): String
}
